enhancements/insights/config-api.md

```diff
@@ -0,0 +1,296 @@
+This would also allow to expose the basic Insights analysis results in the corresponding status section. This functionality now relies on the in-cluster Prometheus (and it will probably not be fully available in the HyperShift architecture).
+Finally, one of the reasons for this enhancement is to define a new way for a user to trigger the Insights data collection.  
+
+### Goals
+
+The goal of this enhancement is to propose a basic structure for a new Insights configuration API and define new configuration options. This includes following:
+
+- Move the majority of the current configuration options to a new configuration API
+- Define a new configuration option providing a way to trigger on demand Insights gathering
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	Gathering Gathering `json:"gathering,omitempty"`
+	// SimpleContentAcces spec attribute includes all the configuration options related to
+	// periodical downloading of the SCA certificates.
+	SimpleContentAccess SimpleContentAccess `json:"simpleContentAccess,omitempty"`
+	// ClusterTransfer spec attribute includes all the configuration options related to
+	// periodical check of pending cluster transfers.
+	ClusterTransfer ClusterTransfer `json:"clusterTransfer,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus GatheringStatus `json:"gatheringStatus,omitempty"`
+	// SimpleContentAccessStatus provides basic information about the SCA controller.
+	SimpleContentAccessStatus SimpleContentAccessStatus `json:"simpleContentAccessStatus,omitempty"`
+	// ClusterTransferStatus provides basic information about the ClusterTransfer controller.
+	ClusterTransferStatus ClusterTransferStatus `json:"clusterTransferStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"insightsReport,omitempty"`
+}
+
+type Gathering struct {
+	// Interval defines the time frequency of gathering. The duration is 2h by default.
+	// +kubebuilder:default=2h
+	Interval time.Duration `json:"interval,omitempty"`
+	// UploadEndpoint defines ingress endpoint for uploading Insights archive
+	// +kubebuilder:default=https://cloud.redhat.com/api/ingress/v1/upload
+	UploadEndpoint string `json:"endpoint,omitempty"`
```

Comment:
is this HTTPS endpoint? or IP address? or hostname?

Reply:
also all the omitempty tags here looks odd ;-)

Reply:
HTTPs endpoint. OK I guess it doesn't make sense to have fields with the default values and `omitempty` tag right.

Reply:
Hopefully it makes sense now.